clusterous/cluster.py

```diff
@@ -735,33 +919,61 @@ def init_cluster(self, cluster_name, cluster_spec, nodes_info=[], logging_level=
+    
+                logging_res = conn.run_instances(defaults.central_logging_ami_id, 
+                                                 min_count=1,
+                                                 key_name=c['key_pair'], 
+                                                 instance_type=defaults.central_logging_instance_type,
+                                                 block_device_map=block_devices, 
+                                                 subnet_id=private_subnet.id, 
+                                                 security_group_ids=[private_security_group.id])
                 logging_tags = {'Name': defaults.central_logging_name_format.format(cluster_name),
                                 defaults.instance_node_type_tag_key: defaults.central_logging_name_tag_value}
                 logging_tags_and_res = [('central-logging', logging_tags, logging_res.instances)]
-
-
+    
+    
             # Wait for controller to launch
             controller = self._wait_and_tag_instance_reservations(controller_tags_and_res)
-
+    
+            # Setup ssh port forwarding on NAT instance
+            self._nat_ssh_port_forwarding(nat_instance.ip_address, controller_instance.private_ip_address)
+    
             # Add controller IP to cluster info file
-            self._set_cluster_info({'controller_ip': controller.values()[0].public_ips[0], 'cluster_name': cluster_name})
+            self._set_cluster_info({'controller_ip': nat_instance.ip_address, 'cluster_name': cluster_name})
             controller_inventory = os.path.expanduser(defaults.current_controller_ip_file)
-            self._write_to_hosts_file(controller_inventory, [controller.values()[0].public_ips[0]], 'controller', overwrite=True)
-
+    
+            self._write_to_hosts_file(controller_inventory, ['{0}:22000'.format(nat_instance.ip_address)], 'controller', overwrite=True)
+    
+            # Shared volume
+            if shared_volume_id:
+                try:
+                    shared_volume = conn.get_all_volumes([shared_volume_id])[0]
+                    if shared_volume.status != 'available':
+                        raise ClusterException('Volume "{0}" is not available'.format(shared_volume_id))
+                    vpc_conn = boto.vpc.connect_to_region(c['region'], aws_access_key_id=c['access_key_id'],aws_secret_access_key=c['secret_access_key'])
+                    if private_subnet.availability_zone != shared_volume.zone:
+                        raise ClusterException('Conflict in availability zone. Subnet "{0}" in "{1}" and Volume "{2}" in "{3}"'.format(private_subnet.id,
+                                                private_subnet.availability_zone, shared_volume_id, shared_volume.zone))
+                except boto.exception.EC2ResponseError as e:
+                    raise ClusterException('Volume "{0}" does not exist'.format(shared_volume_id))
+    
```

Comment:
Ideally this check should occur as early as possible, ie before the NAT is created. With this check done here, if the user specifies a volume that is in the wrong subnet, the error will occur after all instances have already been launched, but before they have been tagged.